ANSWERS.md

```diff
@@ -0,0 +1,15 @@
+What are PropTypes used for? Please describe why it's important to type check our data in JavaScript.
+
+Prop Types are used for typechecking, mostly. As your app grows there are bound to be errors in your data. Prop types help to catch them with validation.
+
+Describe a life-cycle event in React?
+
+A life-cycle event in React is an event when the page loads. For example, there are 3 phases. Render phase, pre-commit phase and commit phase. The render phase consists of the constructor and render functions. The pre-commit phase consists of events that can read the DOM (ex: getSnapshotBeforeUpdate). Commit phase consists of events that can work with the DOM, run side effects and schedule updates (ex: componentDidMount). Additionally there is Mounting, Updating and Unmounting. You use setState() during the updating phase. And componentWillUnmount in the unmounting phase. 
+
+Explain the details of a Higher Order Component?
+
+It's a function that takes a component and returns a new component. 
```

Comment:
Interesting tidbit:

"Note that a HOC doesn’t modify the input component, nor does it use inheritance to copy its behavior. Rather, a HOC composes the original component by wrapping it in a container component. A HOC is a pure function with zero side-effects."

https://reactjs.org/docs/higher-order-components.html